Doc-tests fail to compile on MacOS on Rust 1.64.0

**Expected Behavior**: Running `cargo test` should succeed

**Observed Behavior**: Running `cargo test` fails

OS: MacOS Monterrey 12.6
Rust toolchain: 1.64.0
Git commit: commit 618e75b693fbd65b6555a650040fe6ccf4a724a8 (HEAD -> main, origin/main, origin/HEAD)

```bash
~/Development/open_source/cucumber-expressions ⑂main  cargo test

    Finished test [unoptimized + debuginfo] target(s) in 0.04s
     Running unittests src/lib.rs (target/debug/deps/cucumber_expressions-b8f4511acfa262b2)

running 71 tests
test combinator::escaped0_spec::matches_normal ... ok
test combinator::escaped0_spec::errors_on_escaped_non_reserved ... ok
test combinator::escaped0_spec::matches_normal_followed_by_escaped_then_normal ... ok
test combinator::escaped0_spec::matches_escaped_followed_by_normal ... ok
test combinator::escaped0_spec::errors_on_ending_with_control_char ... ok
test combinator::escaped0_spec::matches_empty ... ok
test combinator::escaped0_spec::matches_escaped_followed_by_normal_then_escaped ... ok
test combinator::escaped0_spec::matches_normal_followed_by_escaped ... ok
test combinator::escaped0_spec::matches_only_escaped ... ok
test parse::spec::alternation::errors_without_slash ... ok
test parse::spec::alternation::fails_on_empty_alternation ... ok
test parse::spec::alternative::errors_on_empty ... ok
test parse::spec::alternation::fails_on_only_optional ... ok
test parse::spec::alternative::escaped_spaces ... ok
test parse::spec::alternative::fails_on_escaped_non_reserved ... ok
test parse::spec::alternative::fails_on_unfinished_optional ... ok
test parse::spec::alternative::not_captures_unescaped_whitespace ... ok
test parse::spec::alternation::basic ... ok
test parse::spec::alternative::optional ... ok
test parse::spec::alternative::text ... ok
test parse::spec::expression::does_not_allow_alternation_in_optional ... ok
test parse::spec::expression::does_not_allow_alternation_with_empty_alternative ... ok
test parse::spec::expression::does_not_allow_alternation_with_empty_alternative_by_adjacent_left_parameter ... ok
test parse::spec::expression::does_not_allow_alternation_with_empty_alternative_by_adjacent_optional ... ok
test parse::spec::expression::allows_escaped_optional_parameter_types ... ok
test parse::spec::alternation::with_optionals ... ok
test parse::spec::expression::does_not_allow_alternation_with_empty_alternative_by_adjacent_right_parameter ... ok
test parse::spec::expression::does_not_allow_empty_optional ... ok
test parse::spec::expression::does_not_allow_nested_optional ... ok
test parse::spec::expression::does_not_allow_optional_parameter_types ... ok
test parse::spec::expression::does_not_allow_parameter_name_with_reserved_characters ... ok
test parse::spec::alternation::with_more_optionals ... ok
test parse::spec::expression::does_allow_parameter_adjacent_to_alternation ... ok
test parse::spec::expression::does_not_allow_unfinished_parenthesis_1 ... ok
test parse::spec::expression::does_not_allow_unfinished_parenthesis_2 ... ok
test parse::spec::expression::does_not_allow_unfinished_parenthesis_3 ... ok
test parse::spec::expression::err_on_escaped_end_of_line ... ok
test parse::spec::expression::allows_parameter_type_in_alternation ... ok
test parse::spec::expression::empty ... ok
test parse::spec::expression::matches_anonymous_parameter_type ... ok
test parse::spec::optional::basic ... ok
test parse::spec::optional::errors_on_empty ... ok
test parse::spec::optional::fails_on_empty ... ok
test parse::spec::optional::fails_on_alternation ... ok
test parse::spec::optional::fails_on_escaped_non_reserved ... ok
test parse::spec::optional::fails_on_nested ... ok
test parse::spec::expression::matches_doubly_escaped_slash ... ok
test parse::spec::optional::fails_on_parameter ... ok
test parse::spec::optional::fails_on_unfinished ... ok
test parse::spec::optional::fails_on_unescaped_reserved_char ... ok
test parse::spec::optional::with_closing_brace ... ok
test parse::spec::optional::with_emoji ... ok
test parse::spec::expression::parameters_ids ... ok
test parse::spec::optional::with_escaped ... ok
test parse::spec::expression::matches_doubly_escaped_parenthesis ... ok
test parse::spec::expression::matches_alternation ... ok
test parse::spec::optional::with_spaces ... ok
test parse::spec::parameter::empty ... ok
test parse::spec::parameter::errors_on_empty ... ok
test parse::spec::parameter::fails_on_escaped_non_reserved ... ok
test parse::spec::expression::matches_optional_before_alternation ... ok
test parse::spec::parameter::fails_on_nested ... ok
test parse::spec::parameter::fails_on_optional ... ok
test parse::spec::parameter::fails_on_unfinished ... ok
test parse::spec::parameter::fails_on_unescaped_reserved_char ... ok
test parse::spec::parameter::named ... ok
test parse::spec::expression::matches_optional_in_alternation ... ok
test parse::spec::parameter::named_with_closing_paren ... ok
test parse::spec::parameter::named_with_emoji ... ok
test parse::spec::parameter::named_with_escaped ... ok
test parse::spec::parameter::named_with_spaces ... ok

test result: ok. 71 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests cucumber-expressions

running 1 test
test src/lib.rs - (line 30) ... FAILED

failures:

---- src/lib.rs - (line 30) stdout ----
error[E0599]: no function or associated item named `regex` found for struct `Expression<_>` in the current scope
 --> src/lib.rs:33:22
  |
6 | let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
  |                      ^^^^^ function or associated item not found in `Expression<_>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
Couldn't compile the test.

failures:
    src/lib.rs - (line 30)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.05s

error: test failed, to rerun pass '--doc'
```

> I think just #![cfg(feature = "into-regex")] would be enough

```rust
#[cfg(feature = "into-regex")] 
fn main() {
    let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
    let caps = re.captures("I have 42 cucumbers in my belly").unwrap();
    
    assert_eq!(&caps[0], "I have 42 cucumbers in my belly");
    assert_eq!(&caps[1], "42");
} 
```

Results in an error:


```
test src/lib.rs - (line 30) ... FAILED

failures:

---- src/lib.rs - (line 30) stdout ----
error[E0601]: `main` function not found in crate `rust_out`
  --> src/lib.rs:40:2
   |
12 | }
   |  ^ consider adding a `main` function at the crate level

error: aborting due to previous error

For more information about this error, try `rustc --explain E0601`.
Couldn't compile the test.

failures:
    src/lib.rs - (line 30)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.04s

error: test failed, to rerun pass '--doc'
```

- https://github.com/rust-lang/rust/issues/85239

> With this I can imagine another issue being reported about crate level docs not being present when running `cargo doc` command 😄

I don't think that practically anyone would care.

--- a/README.md
+++ b/README.md
@@ -16,11 +16,13 @@ This crate provides [AST] parser, and [`Regex`] expansion of [Cucumber Expressio
 ```rust
 use cucumber_expressions::Expression;
 
-let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
-let caps = re.captures("I have 42 cucumbers in my belly").unwrap();
-
-assert_eq!(&caps[0], "I have 42 cucumbers in my belly");
-assert_eq!(&caps[1], "42");
+#[cfg(feature = "into-regex")] {
+    let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
+    let caps = re.captures("I have 42 cucumbers in my belly").unwrap();
+    
+    assert_eq!(&caps[0], "I have 42 cucumbers in my belly");
+    assert_eq!(&caps[1], "42");
+} 
 ```
 
 

--- a/README.md
+++ b/README.md
@@ -14,15 +14,16 @@ Rust implementation of [Cucumber Expressions].
 This crate provides [AST] parser, and [`Regex`] expansion of [Cucumber Expressions].
 
 ```rust
+# // `Regex` expansion requires `into-regex` feature.
+# #[cfg(feature = "into-regex")] {
 use cucumber_expressions::Expression;
 
-#[cfg(feature = "into-regex")] {
-    let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
-    let caps = re.captures("I have 42 cucumbers in my belly").unwrap();
-    
-    assert_eq!(&caps[0], "I have 42 cucumbers in my belly");
-    assert_eq!(&caps[1], "42");
-} 
+let re = Expression::regex("I have {int} cucumbers in my belly").unwrap();
+let caps = re.captures("I have 42 cucumbers in my belly").unwrap();
+
+assert_eq!(&caps[0], "I have 42 cucumbers in my belly");
+assert_eq!(&caps[1], "42");
+# }
 ```
 
 

